programs/openbook-v2/src/instructions/prune_orders.rs

```diff
@@ -0,0 +1,33 @@
+use anchor_lang::prelude::*;
+
+use crate::accounts_ix::*;
+use crate::error::*;
+use crate::state::*;
+
+// TODO
+#[allow(clippy::too_many_arguments)]
+pub fn prune_orders(ctx: Context<PruneOrders>, limit: u8) -> Result<()> {
+    let mut account = ctx.accounts.open_orders_account.load_full_mut()?;
+    let market = ctx.accounts.market.load()?;
+
+    // check market is expired
+    require!(
+        market.time_expiry == -1 || market.time_expiry < Clock::get()?.unix_timestamp,
+        OpenBookError::MarketHasNotExpired
+    );
+    let close_admin =
```

Comment:
not sure if this (as well as `close_market`) should be permisionless, so any expired market could be closed

Reply:
the point is to get the rent, and permisionless is a wide topic. you need an admin to close it, the admin who will get the rent

Reply:
I was thinking that in that case, the rent could go to whoever closes the market. but if that's not going to be supported, maybe it makes sense to add a constraint that to have expiration, and admin needs to exist?